tests/unit/activity.test.ts

```diff
@@ -372,10 +366,41 @@ describe("Activity", () => {
           expect(error.getAgreement()).toBeDefined();
           expect(error.getProvider()?.name).toEqual("Test Provider");
           expect(error.previous?.toString()).toEqual(
-            "Error: Command #0 getExecBatchResults error: GSB error: remote service at `test` error: GSB failure: Bad request: endpoint address not found",
+            "Error: Failed to fetch activity results. Attempt: 1. Error: non-retryable error",
           );
-          expect(error.toString()).toEqual(
-            "Error: Unable to get activity results. Command #0 getExecBatchResults error: GSB error: remote service at `test` error: GSB failure: Bad request: endpoint address not found",
+          return res();
+        });
+        results.on("data", () => null);
+      });
+    });
+
+    it("should handle retryable error", async () => {
+      const activity = await Activity.create(instance(mockAgreement), instance(mockYagna), {
+        activityExeBatchResultPollIntervalSeconds: 10,
+      });
+      const command1 = new Deploy();
+      const command2 = new Start();
+      const command3 = new Run("test_command1");
+      const script = Script.create([command1, command2, command3]);
+      const results = await activity.execute(script.getExeScriptRequest(), false, 1_000, 1);
+
+      const error = {
+        message: "timeout",
+        status: 408,
+        toString: () => `Error: timeout`,
+      };
+
+      when(mockActivityControl.getExecBatchResults(anything(), anything(), anything(), anything())).thenReject(error);
+
+      return new Promise<void>((res) => {
+        results.on("error", (error: GolemWorkError) => {
+          expect(error).toBeInstanceOf(GolemWorkError);
+          expect(error.code).toEqual(WorkErrorCode.ActivityResultsFetchingFailed);
+          expect(error.getActivity()).toBeDefined();
+          expect(error.getAgreement()).toBeDefined();
+          expect(error.getProvider()?.name).toEqual("Test Provider");
+          expect(error.previous?.toString()).toEqual(
+            "Error: Failed to fetch activity results. Attempt: 2. Error: timeout",
```

Comment:
I'm not 100% sure what is this test actually testing? Is it the "Attempt: 2" line instead of "Attempt 1"? What do you think about checking how many times `.getExecBatchResults` was called? Or making it so the first time it throws and second time returns results, and we expect results (because it was retried)?

Reply:
you're right, I can check it better